### DISCUSSION TITLE
Lab2 C synthesis ERROR

### DISCUSSION THREAD

**Discussion by JIAJIA4** [bol-edu/HLS-SOC-Discussions#78] [Q&A] [answered]
在vitis_hls執行FIRN11Stream的C synthesis時遇到下列Error，如附圖，
ERROR: [HLS 214-208] The ap_axis|ap_axiu|qdma_axis|hls::axis data types must only be used for AXI-Stream ports in the interface. 'pstrmOutput' is not an AXI-Stream and/or is not a port in the interface

![error](https://github.com/bol-edu/HLS-SOC-Discussions/assets/145210303/1bf4f0bc-1f6c-4ea1-b34c-080801ec72be)

請問該怎麼解決呢？謝謝

**Comment by hbx1241** [comment, marked as answer]
不確定是不是你的 pragma 設定有問題？（我看wraning strm 的 pragma 好像被忽略了）
我用範例預設的可以跑
![2023-09-29_16-07](https://github.com/bol-edu/HLS-SOC-Discussions/assets/57489184/bd744458-2ff6-451e-995e-554607a6b57d)

**Reply by JIAJIA4** [reply]
謝謝，我原本以為Stream也要改成跟maxi一樣的pragma，原來不用改就好！
